Add examples of good comments and mention punctuation (#130)
tidyverse/style: functions.Rmd

In code, use comments to explain the "why" not the "what" or "how". Each line 
of a comment should begin with the comment symbol and a single space: `# `.

```{r, eval = FALSE}
# Good

# Objects like data frames are treated as leaves
x <- map_if(x, is_bare_list, recurse)

# Bad

# Recurse only with bare lists
x <- map_if(x, is_bare_list, recurse)
```

> Comments are capitalised and don't end with a full stop, unless they

Reviewer: ```suggestion
Comments should be in sentence case, and only end with a full stop if they
```
